server/base/src/main/java/org/apache/accumulo/server/init/FileSystemInitializer.java

```diff
@@ -194,28 +215,24 @@ private void createMetadataFile(VolumeManager volmanager, String fileName,
         .forFile(file, fs, fs.getConf(), cs).withTableConfiguration(conf).build();
     tabletWriter.startDefaultLocalityGroup();
 
+    TreeMap<Key,Value> sorted = new TreeMap<>();
+    for (InitialTablet initialTablet : initialTablets) {
+      // sort file contents in memory, then play back to the file
+      for (Map.Entry<Key,Value> entry : initialTablet.createEntries().entrySet()) {
+        sorted.putAll(initialTablet.createEntries());
+      }
```

Comment:
This seems to be putting all the entries for each entry doing redundant work.

```suggestion
     sorted.putAll(initialTablet.createEntries());
```